samcli/commands/publish/command.py

```diff
@@ -31,31 +33,40 @@
 """
 SHORT_HELP = "Publish a packaged AWS SAM template to the AWS Serverless Application Repository."
 SERVERLESSREPO_CONSOLE_URL = "https://console.aws.amazon.com/serverlessrepo/home?region={}#/published-applications/{}"
+SEMANTIC_VERSION_HELP = "Optional. The value provided here overwrites SemanticVersion in the template metadata."
```

Comment:
are there any restrictions on what the version can be set to? or is it up to the user to setup any version?

Reply:
It's up to the user, but if they pass in invalid version numbers, the API call would fail. I didn't include an explanation of valid version numbers here, because it's already in the [metadata doc](http://dev-dsk-mhirayam-2c-5d259f39.us-west-2.amazon.com/docs-preview/serverless-application-model/latest/developerguide/serverless-sam-template-publishing-applications-metadata-properties.html).

Reply:
Do we catch the exception correct and provide a useful message to customers within the CLI?

Reply:
Will add code to catch the exception and print user-friendly error message.